File: frontend/voting-system/src/api/votingService.ts
import api from './config'

export const votingService = {

  // 獲取所有投票項目
  getAllVoteItems() {
    return api.get('/voting/items')
  },

  // 新增投票項目
  addVoteItem(itemName: string) {
    return api.post('/voting/items', itemName)
  },

  // 更新投票項目
  updateVoteItem(itemId: number, itemName: string) {
    return api.put(`/voting/items/${itemId}`, itemName)
  },

  // 刪除投票項目
  deleteVoteItem(itemId: number) {
    return api.delete(`/voting/items/${itemId}`)
  },

  // 新增投票記錄（複選）
  addVoteRecords(voteItemIds: number[]) {
    return api.put('/voting/records', {
        voteItemIds: voteItemIds
      })
  },

  // 獲取用戶投票記錄
  getVoteRecordsByUserId() {
    return api.get('/voting/records/user/')
  }

}

File: backend/voting-system-spring-vuejs/src/main/java/com/example/votingSystem/service/VotingService.java
package com.example.votingSystem.service;

import java.util.List;

import com.example.votingSystem.model.dto.VoteItemDto;
import com.example.votingSystem.model.dto.VoteRecordDto;

public interface VotingService {

	public List<VoteItemDto> getAllVoteItems();
	
	public void addVoteItem(String itemName);
	
	public void updateVoteItem(Long itemId, String newName);
	
	public void deleteVoteItem(Long itemId);
	
	public void addVoteRecord(Long userId, List<Long> voteItemIds);
	
	public List<Object[]> getVoteRecordsByUserId(Long userId);
	
}
